### DISCUSSION TITLE
When to use `Backpex.Filters.Boolean` vs `Backpex.Filters.MultiSelect`

### DISCUSSION THREAD

**Discussion by John-Goff** [naymspace/backpex#631] [Q&A] [answered]
From the docs, both `Boolean` and `MultiSelect` filters render a list of checkboxes for each given option. What's the intended use case of one vs the other? They seemingly overlap, though I could be misunderstanding. 

## Context

I have an `Ecto.Enum` array field for `:categories` on my schema, users can select one or multiple categories during sign up. I'd like for admins to be able to search for users who have selected one of the searched for categories. I've currently implemented this with a `Backpex.Filters.Boolean` filter, though I tried a `Backpex.Filters.MultiSelect` which also worked the way I want. The `Boolean` filter had a better UI (less clicks), and less code in the module, so I went with that. I just feel like I'm missing something, which is what led to this question.

**Comment by Flo0807** [comment, marked as answer]
Hey! I think the docs are a little misleading here. While both filters render a list of checkboxes, the `MultiSelect` filter wraps these inputs into a select input that you can open and close. I guess the difference between these two filters is mostly in the UI.

I would suggest using the `Boolean` filter if you have a handful of predefined values (e.g. a published status of a post or a list of predefined categories). Once you start loading options dynamically, e.g. if you want to filter for specific users, I recommend using the `MultiSelect` filter, as you can open and close it.

We started with the Boolean filter. Then we had a requirement where we needed to filter for specific users in a select input. We decided to add a different filter to support this because the UI is fundamentally different.

I guess we should do a better job of highlighting the differences in our documentation! Thanks for your discussion.
